bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/LedgerMetadataIndex.java

```diff
@@ -345,6 +345,7 @@ public void removeDeletedLedgers() throws IOException {
                 log.debug("Persisting deletes of ledgers {}", deletedLedgers);
             }
 
+            pendingDeletedLedgers.clear();
```

Comment:
Thanks for your fix.  But I  think  we should call   `pendingDeletedLedgers.remove(ledgerId)`  after line342.
`pendingDeletedLedgers` cloud being adding when call `clear()`,  cause some ledgers not have  deleted form ledgersDB but  clear from `pendingDeletedLedgers `.

By the way.  `pendingDeletedLedgers ` is a `Sets.newConcurrentHashSet()`.   I think it will not concurrent error.

Reply:
that's ok